state/txindex/kv/utils.go

```diff
@@ -66,16 +66,18 @@ func ParseEventSeqFromEventKey(key []byte) (int64, error) {
 
 func dedupHeight(conditions []query.Condition) (dedupConditions []query.Condition, heightInfo HeightInfo) {
 	heightInfo.heightEqIdx = -1
+	found := false
 	heightRangeExists := false
 	var heightCondition []query.Condition
 	heightInfo.onlyHeightEq = true
 	heightInfo.onlyHeightRange = true
 	for _, c := range conditions {
 		if c.CompositeKey == types.TxHeightKey {
 			if c.Op == query.OpEqual {
-				if heightRangeExists {
+				if heightRangeExists || found {
```

Comment:
Do we need to back/forward port this?

Reply:
IT has to be included in the open PR for 0.37. I have it in my todo for that PR. As well as porting the UTs you added to this PR.